transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java

```diff
@@ -272,27 +275,31 @@ private StructuredRecord createStructuredRecord(@Nullable CSVRecord record, Stru
     @Description("Specifies the schema that has to be output.")
     private final String schema;
 
-    public Config(String format, Character delimiter, String field, String schema) {
-      this.format = format;
+    public Config(@Nullable String format, @Nullable Character delimiter, String field, String schema) {
+      this.format = format == null ? "DEFAULT" : format;
       this.delimiter = delimiter;
       this.field = field;
       this.schema = schema;
     }
 
     private void validate() {
-      // Check if the format specified is valid.
-      if (format == null || format.isEmpty()) {
-        throw new IllegalArgumentException("Format is not specified. Allowed values are DEFAULT, EXCEL, MYSQL," +
-                                             " RFC4180, Pipe Delimited, Tab Delimited or Custom");
+
+      if (format == null) {
+        if (delimiter != null && delimiter != 0) {
+          throw new IllegalArgumentException("Custom delimiter can only be used for format option 'Custom'.");
+        }
+        return;
       }
 
       // Check if format is one of the allowed types.
       if (!format.equalsIgnoreCase("DEFAULT") && !format.equalsIgnoreCase("EXCEL") &&
         !format.equalsIgnoreCase("MYSQL") && !format.equalsIgnoreCase("RFC4180") &&
         !format.equalsIgnoreCase("Tab Delimited") && !format.equalsIgnoreCase("Pipe Delimited")
-        && !format.equalsIgnoreCase("Custom")) {
-        throw new IllegalArgumentException("Format specified is not one of the allowed values. Allowed values are " +
-                                             "DEFAULT, EXCEL, MYSQL, RFC4180, Pipe Delimited, Tab Delimited or Custom");
+        && !format.equalsIgnoreCase("Custom") && !format.equalsIgnoreCase("PDL") &&
+        !format.equalsIgnoreCase("TDF")) {
+        throw new IllegalArgumentException(String.format("Format %s is not one of the allowed values. Allowed values " +
```

Comment:
better to keep a set of allowed formats. Then you can check that the format is in the set, and do `Joiner.on(',').join(FORMATS)` here and won't have to worry about forgetting to update this message if a new format is added or removed